activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -39,20 +39,23 @@ def set_fixture_class(class_names = {})
       end
 
       def fixtures(*fixture_set_names)
+        env = ActiveRecord::ConnectionHandling::DEFAULT_ENV.call
+        db_config = ActiveRecord::Base.configurations.configs_for(env_name: env, spec_name: "primary")
```

Comment:
I've essentially done everything up until the iterating over each DB config and doing the connection switching in this PR. This is because https://github.com/rails/rails/pull/33945 was getting really complicated and I tried to break things up into smaller pieces (as we had previously discussed).

Only using the primary DB config here as a temporary. I want to make sure we agree with the approach up to this point so we can finish up by addressing the connection switching roadblocks in https://github.com/rails/rails/pull/33945.

Does that make sense?